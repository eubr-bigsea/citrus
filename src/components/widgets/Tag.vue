<template>
    <div v-if="readOnly">
        {{selected}}
    </div>
    <div v-else>
        <LabelComponent :field="field" :value="value"></LabelComponent>aa
        <b-form-tags @input="updatedTag" v-model="selected" name="field.name" class="lemonade-tag-input pt-3"
            placeholder="" add-button-text="+" :input-type="inputType" />
    </div>
</template>
<script>
    import LabelComponent from './Label.vue'
    import Widget from '../../mixins/Widget.js';
    export default {
        mixins: [Widget],
        components: { LabelComponent },
        mounted() {
            this.$root.$emit(this.message,
                this.field, this.value || this.field['default']);
        },
        computed: {
            selected: {
                get() {
                    if (this.field.multiplicity === 2 || this.field.multiplicity === 3) {
                        this.tags = [];
                        if (this.value) {
                            try {
                                this.tags = JSON.parse(this.value);
                            } catch (e) {
                                console.log(e);
                            }
                        }
                        return this.tags;
                    } else {
                        return this.value || this.field.default;
                    }
                },
                set(value) {
                }
            },
            inputType() {
                return this.field.data_type || 'text';
            }
        },
        methods: {
            updatedTag(values) {
                this.$root.$emit(this.message, this.field, values, values);

            },
        },
        ready: function () {
            if (this.field['default'] && (this.value === null || this.value === '')) {
                this.value = this.field['default'];
            }
        },
    }

</script>
<style scoped>
    /* Chrome, Safari, Edge, Opera */
    .lemonade-tag-input>>>input::-webkit-outer-spin-button,
    .lemonade-tag-input>>>input::-webkit-inner-spin-button {
        -webkit-appearance: none;
        margin: 0;
    }

    /* Firefox */
    .lemonade-tag-input>>>input[type=number] {
        -moz-appearance: textfield;
    }
</style>