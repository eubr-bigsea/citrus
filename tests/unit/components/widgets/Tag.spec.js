import { shallowMount } from '@vue/test-utils';
import Tag from '$SRC/components/widgets/Tag.vue';
import LabelComponent from '$SRC/components/widgets/Label.vue';
import { expect, it } from 'vitest';

describe('Tag.vue', () => {
  it('ReadOnly true', () => {
    const wrapper = shallowMount(Tag, {
      propsData: {
        readOnly: true,
        value: 'Abcd',
      },
    });
    expect(wrapper.text()).toBe('Abcd');
  });

  it('ReadOnly false', () => {
    const wrapper = shallowMount(Tag, {
      propsData: {
        readOnly: false,
        field: { name: 'FieldName' },
        value: 'SomeValue',
      },
    });
    expect(wrapper.findComponent(LabelComponent).exists()).toBe(true);
  });

  it('snapshots the component', () => {
    const wrapper = shallowMount(Tag, { propsData: { field: {} } });
    expect(wrapper.text()).toMatchSnapshot();
});
});


