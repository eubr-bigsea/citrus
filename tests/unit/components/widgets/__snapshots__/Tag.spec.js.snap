// Vitest Snapshot v1

exports[`Tag.vue > snapshots the component 1`] = `"aa"`;
